### DISCUSSION TITLE
Install non-python dependency

### DISCUSSION THREAD

**Discussion by karol-bisztyga** [python-poetry/poetry#7285] [Q&A]
Hey!
I've been playing with this for a while and couldn't figure out anything satisfying, unfortunately.
The problem is, I have a python project in which I use poetry. I have some python dependencies but I also wanted to add a rust project there and use [pyo3](https://github.com/PyO3/pyo3) to connect them (I mean, use rust from python). 
I managed to turn the rust lib into the python lib by adding some dummy files like `pyproject.toml`/`setup.cfg`. But I still have to use bash automation for this as poetry doesn't seem to be flexible enough to handle such a setup - I need to run some commands:
```
poetry run rustup override set nightly
poetry run pip install maturin
poetry run maturin develop --release
```
I also tried to play with `setup.py` but with no good results.

Is it possible to handle this using poetry somehow?

Thanks in advance!

**Comment by laurentS** [comment]
I'm actually interested in something like this, with a similar use case where my python code relies on a 3rd party binary tool that I just need to download to a specific location so the python code can run it. I've built automation outside python to do this, but it means the tool isn't really self contained, so it doesn't feel very clean.

@karol-bisztyga for your python+rust use case, have you looked at https://www.maturin.rs/ ? I haven't used it myself, but I've heard good things about it.

**Comment by dimbleby** [comment]
poetry is not a general pupose task runner and this is out of scope for the project.

there might be plugins that would help eg I've sometimes seen mention of https://github.com/nat-n/poethepoet (neither a recommendation nor an un-recommendation, I've never used it).  

personally I'd probably write a makefile
